### PR TITLE
Release v2.0.0 with updated TBprofiler and new NTMprofiler results

### PR DESCRIPTION
**Pull request by abhi18av**
This PR 

1. Update `tb-profiler` to `v6.2.1` along with the `WHO mutations catalog v2` with the database version `30f8bc37df15affa378ebbfbd3e1eb4c5903056e`
 
2. Addition of `ntm-profiler` 

3. Two new `conda-envs` and containers have been added to the pipeline, relying directly on upstream biocontainers for both `ntm-profiler` and `tb-profiler`

4. Changes to the directory structure

5. Addition of mixed-infections resistance summary script

6. Addition of `low_memory` profile to accommodate lower end infrastructure

7. Updates to the names of parameters for triggering partial/optional workflows

8. Improve the codebase of summarization scripts, simplication and publications of json data

9. Tighten the retry strategy for failing FASTQs

10. Publish the validated samplesheet to the results folder.

### CODE REVIEW

**Comment by vrennie** [comment]
HI @abhi18av, we realised that the structural and minor variant workflows now run on all samples, regales of whether they meet the NTM and coverage thresholds. This should not be the case. We should run the samples on all samples except those who no not meet the following thresholds:
1. MAPPED_NTM_FRACTION_16S_THRESHOLD_MET
2. COVERAGE_THRESHOLD_MET
3. BREADTH_OF_COVERAGE_THRESHOLD_MET

Could you please implement these changes ASAP? Thanks

**Comment by vrennie** [comment]
HI @abhi18av, just realised that what I suggested may not be possible because Nextflow is reading the passed.tsv and failed.tsv, is is a better option for @mdediegofuertes to add filters in the summarise_resistance.py to grab the correct samples and only generate summaries for those samples?

**Review by vrennie — APPROVED** [review]

File: bin/summarize_resistance_mixed_infection.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def create_resistance_df(sample_res, method):
 #===============
     stats_df = pd.read_csv(args["merged_cohort_stats_file"], sep="\t")
     mixed_inf_stats_df = stats_df[stats_df["RELABUNDANCE_THRESHOLD_MET"] == 0]
-    filtered_stats_df = mixed_inf_stats_df[(mixed_inf_stats_df[["MAPPED_NTM_FRACTION_16S_THRESHOLD_MET", "COVERAGE_THRESHOLD_MET", "BREADTH_OF_COVERAGE_THRESHOLD_MET"]] == 1).all(axis=1)]
+    filtered_stats_df = mixed_inf_stats_df.loc[ (mixed_inf_stats_df["RELABUNDANCE_THRESHOLD_MET"]==0) & (mixed_inf_stats_df["MAPPED_NTM_FRACTION_16S_THRESHOLD_MET"]==1) & (mixed_inf_stats_df["COVERAGE_THRESHOLD_MET"]==1) & (mixed_inf_stats_df["BREADTH_OF_COVERAGE_THRESHOLD_MET"]==1)]
```
Comment (abhi18av) [inline bin/summarize_resistance_mixed_infection.py:139]: @vrennie @mdediegofuertes , the issue was hidden in this refactor here regarding the NTM samples not showing up.

File: main.nf [hunk 2]
```diff
@@ -116,8 +116,8 @@ workflow {
 
         REPORTS_WF( QUALITY_CHECK_WF.out.reports_fastqc_ch,
                     UTILS_MERGE_COHORT_STATS.out.merged_cohort_stats_ch,
-                    MINOR_VARIANTS_ANALYSIS_WF.out.minor_variants_results_ch,
                     MERGE_WF.out.major_variants_results_ch,
+                    MINOR_VARIANTS_ANALYSIS_WF.out.minor_variants_results_ch,
```
Comment (abhi18av) [inline main.nf:120]: Beyond the tweak the `summarize_resistance` script, the order was reversed for passing the `major` and `minor` variants 😓

**Pull request merged**